[docs] what is rendered when for #await

benmccann opened the pull request:
I was looking at https://github.com/sveltejs/kit/issues/2520 and was a bit unclear on how `{#await}` works exactly. @Conduitry filled me in on some missing details that helped elucidate things. I thought they could be useful to add to the docs

Conduitry reviewed (changes requested):
Mentioning hydration here seems weird. There's no guarantee that the user is using hydration at all, and it also hasn't been mentioned anywhere previously on this page.

I'm not sure we need to mention at all that pending state is rendered initially on the client. What else would it be? You can't synchronously get the status of a promise, even one that's already settled.

And saying it's rendered initially on the server seems misleading: it's what's rendered synchronously on the browser, period.

benmccann commented:
I updated it. Is it any better now? (also open to other suggestions)

The fact that rendering happens only synchronously isn't something I was really aware of since that's sort of hidden inside the internals of the framework / plugin and I rarely invoke the renderer directly. Even after being aware of it, I didn't really connect it to the fact that only a part of this language feature would have any affect on rendering on the server and what that meant the behavior would be.

> And saying it's rendered initially on the server seems misleading: it's what's rendered synchronously on the browser, period.

I'm not sure I understood this comment. Do you think what I have updated it to is incorrect?

dummdidumm commented on site/content/docs/02-template-syntax.md:
```diff
@@ -309,7 +309,7 @@ An each block can also have an `{:else}` clause, which is rendered if the list i
 ---
 
-Await blocks allow you to branch on the three possible states of a Promise — pending, fulfilled or rejected.
+Await blocks allow you to branch on the three possible states of a Promise — pending, fulfilled or rejected. If using SSR, the server will render the pending state.
```
Maybe `In SSR-mode` instead?

benmccann replied:
ok. tweaked it a bit
